app/src/main/java/com/concordia/mcga/activities/SplashActivity.java

```diff
@@ -0,0 +1,37 @@
+package com.concordia.mcga.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+
+public class SplashActivity extends AppCompatActivity {
+
+    private final long VIEW_TIME = 2000;
+
+    /**
+     * Creates the splash screen. Switch to MainActivity after 'VIEW_TIME' milliseconds
+     * @param savedInstanceState The savedInstanceState is a reference to a Bundle object that is passed into the onCreate method of every Android Activity
+     */
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.splash_screen);
+
+        Thread mainThread = new Thread(){
+            @Override
+            public void run() {
+                try {
+                    sleep(VIEW_TIME);
```

Comment:
I think it is actually an ok implementation, as the reason our application takes time to boot up is because it is in debug mode. If we build a release version it won't take as much time to boot. Given that, the main thread will be killed to quickly and the user will see the screen flicker. IMO looking at a splash screen for 2 seconds is better than looking at it for 300 ms.

The 2nd point about staring at a blank screen, however, is valid.